components/mitsubishi_uart/mitsubishi_uart-packetprocessing.cpp

```diff
@@ -285,13 +285,34 @@ void MitsubishiUART::processPacket(const StandbyGetResponsePacket &packet) {
   }
 
   //TODO: Not sure what AutoMode does yet
-};
+}
+
 void MitsubishiUART::processPacket(const ErrorStateGetResponsePacket &packet) {
   ESP_LOGV(TAG, "Processing %s", packet.to_string().c_str());
   routePacket(packet);
-  // TODO: The MHK2 thermostat often asks for this, but the response is usually just all zeros.  Could be be checking for
-  // errors / messages / warnings.  Should check when e.g. the filter life runs out.
-};
+
+  auto oldErrorCode = error_code_sensor->raw_state;
+
+  // TODO: Include friendly text from JSON, somehow.
+  if (!packet.errorPresent()) {
+    error_code_sensor->raw_state = "No Error Reported";
+  } else if (auto rawCode = packet.getRawShortCode() != 0x00) {
+    // Not that it matters, but good for validation I guess.
+    if ((rawCode & 0x1F) > 0x15) {
+      ESP_LOGW(TAG, "Error short code %x had invalid low bits. This is an IT protocol violation!", rawCode);
+    }
+
+    error_code_sensor->raw_state = "Error " + packet.getShortCode();
+  } else if (packet.getErrorCode() != 0x8000) {
+    error_code_sensor->raw_state = "Error " + to_string(packet.getErrorCode());
+  } else {
+    // Logic bug, should never happen.
+    ESP_LOGW(TAG, "Packet indicated an error was present, but none of the error states matched. wat.");
+  }
```

Comment:
Rather than having both of these statements with one case just for a "logic bug", let's just do:
```
else {
    error_code_sensor->raw_state = "Error " + to_string(packet.getErrorCode());
}
```
and be done with it.  If it ever reports "Error 8000", we'll know there's an issue, but this is not any worse than not updating the sensor at all, and logging a warning.